### PR TITLE
Add more fields to NIC

### PR DESCRIPTION
**Pull request by jak3kaj**
Added the following to the `NIC` struct:
* `Speed`
* `Duplex`
* `SupportedLinkModes`
* `SupportedPorts`
* `SupportedFECModes`
* `AdvertisedLinkModes`
* `AdvertisedFECModes`

Refactoring PR #335

### CODE REVIEW

File: README.md
Comment (jaypipes) [inline README.md:720]:
```suggestion
* `ghw.NIC.Speed` is a string showing the current link speed.  On Linux, this 
  field will be present even if `ethtool` is not available.
```

File: README.md
Comment (jaypipes) [inline README.md:722]:
```suggestion
* `ghw.NIC.Duplex` is a string showing the current link duplex. On Linux, this 
  field will be present even if `ethtool` is not available.
```

**Review by jaypipes — APPROVED** [review]
Fantastic work on this @jak3kaj, really appreciate your effort! :)

File: pkg/net/net_linux.go
Comment (jaypipes) [inline pkg/net/net_linux.go:283]: @jak3kaj I think you just need to remove this function.

**Review by jaypipes — CHANGES_REQUESTED** [review]

**Review by jaypipes — COMMENTED** [review]
@jak3kaj how's it going? I wanted to know if you think you can fix up this PR today? I was looking to cut a new release of `ghw` today and wanted to get this excellent work into the release :)

I think the only thing you need to do is rebase and remove the now-uncalled functions.

File: pkg/net/net_linux.go
Comment (jak3kaj) [reply pkg/net/net_linux.go]: Yes, I refactored this function out and forgot to remove it.

**Review by jaypipes — APPROVED** [review]
I'm good with this, thank you @jak3kaj! Could probably place that `readFile` helper method into the util pkg long-term but I won't hold this up for that request.

@ffromani if you are good with this after review, please feel free to merge.

**Comment by ffromani** [comment]
> I'm good with this, thank you @jak3kaj! Could probably place that `readFile` helper method into the util pkg long-term but I won't hold this up for that request.
> 
> @ffromani if you are good with this after review, please feel free to merge.

Looking at this ASAP

**Comment by jak3kaj** [comment]
> I'm good with this, thank you @jak3kaj! Could probably place that `readFile` helper method into the util pkg long-term but I won't hold this up for that request.
> 
> @ffromani if you are good with this after review, please feel free to merge.

I'm working on the 3rd PR to add support for the WOL data.  I can move `readFile` to utils there as well.  I got busy with something else, so this will be in a week or so.

**Review by ffromani — COMMENTED** [review]
(VERY LATE to the party but...) LGTM, thanks!

**Pull request merged**
